src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const copyGoods = [...goodsFromServer];
+  const [sortType, setsortType] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  const getReorderedGoods = (goods: string[]) => {
+    goods.sort((g1, g2) => {
```

Comment:
Sort is a mutable method. You don't need to rewrite it to a new constant as you do it below with `let visibleGoods`.  You can just sort a copy of goodsFromServer without an additional function